docs/search/2_cpr_searches.md

```diff
@@ -0,0 +1,313 @@
+}
+```
+
+#### `document_passage` schema
+
+All the same applies to the document_passage schema, but on different fields.
+
+``` js
+field text_block_not_stemmed type string {
+    indexing: input text_block | summary | index
+    stemming: none
+}
+
+rank-profile default_passage inherits default {
+    inputs {
+        query(passage_weight) double: 1.0
+    }
+}
+
+rank-profile exact_not_stemmed inherits default_passage {
+    function text_score() {
+        expression: fieldMatch(text_block_not_stemmed)
+    }
+    first-phase {
+        expression: query(passage_weight) * text_score()
+    }
+    summary-features: query(passage_weight) text_score()
+}
+```
+
+## Hybrid search
+
+Hybrid search is the default search on our tool. It's implementation is similar to exact match search, but with a few extra parts to the query and schema. Here, just the differences will be highlighted.
+
+### Annotated YQL
+
+The group clause is exactly the same as before! But in the retrieval part, instead of using `contains` we use:
+
+- [userInput](https://docs.vespa.ai/en/reference/query-language-reference.html#userinput): Vespa's way of turning a free text query into the lower-level query that it thinks is most suitable. Operates on the *default fieldset* defined in each schema.
+- [nearestNeighbour](https://docs.vespa.ai/en/nearest-neighbor-search.html#querying-using-nearestneighbor-query-operator): performs approximate nearest neighbour search using an embedding of the query and the `text_embedding` field, which is the the vector-encoded version of each text block. This is what makes our search 'semantic'.
+
+``` sql
+select * from sources family_document, document_passage where ( 
+    ---
+    (userInput(@query_string)) 
+    or ( 
+        [{\"targetNumHits\": 1000}] nearestNeighbor(text_embedding,query_embedding) ) 
+    ) 
+    limit 0 
+| all( 
+    group(family_import_id) 
+    output(count()) max(20) 
+    each( output(count()) max(10) each( output( summary(search_summary) ) ) ) 
+)
+```
+
+### Relevant parts of schemas
+
+The ranking for each field is as follows:
+
+- family names are ranked using BM25.
+- family descriptions are ranked using a weighted sum of BM25 and vector similarity. The weight of the latter is set to 0 in the SDK, disabling vector search on descriptions.
+- passages are ranked using a weighted sum of BM25 and vector similarity.
+
+[BM25](https://en.wikipedia.org/wiki/Okapi_BM25) ranks according to the number of times a term in the query appears in each record, penalising occurrences (like 'climate') which appear in lots of records compared to those that are rarer.
+
+``` js
+// family_document hybrid rank profile
+rank-profile hybrid inherits default_family {
+    inputs {
+        query(query_embedding) tensor<float>(x[768])
+        query(description_bm25_weight) double: 1.0
+        // NOTE: this is set to 0 in the SDK, disabling embedding search on descriptions
```

Comment:
Could we explain why it's set to 0 in the SDK, or why it's still `1.0` in the schema? I _think_ I know the reason, but I reckon that might not be clear to newcomers

Reply:
i've [given it a go](https://github.com/climatepolicyradar/cpr-sdk/pull/181/commits/b8fe72aa54fb843bc015dfaac00e10494446fa1d). wdyt?

Reply:
lovely!